test/inductor/test_fused_attention.py

```diff
@@ -357,7 +357,120 @@ def forward(self, query, key, value, attn_mask) -> torch.Tensor:
                 model, args1=args, contains=False, atol=1e-4, has_fuse_pattern=False
             )
 
+    @skipIfRocm
+    def _test_sdpa_rewriter_11(self):
+        def dot_prod_attention(
+            query: torch.Tensor, key: torch.Tensor, value: torch.Tensor
+        ) -> torch.Tensor:
+            """Input tensors assumed to have shape (batch_size, seq_len, n_head, embed_dim)"""
+            q = query.transpose(1, 2)
+            k = key.transpose(1, 2)
+            v = value.transpose(1, 2)
+            return (
+                torch.matmul(q, k.transpose(-2, -1))
+                .div(math.sqrt(key.shape[-1]))
+                .softmax(dim=-1)
+                .matmul(v)
+            )
+
+        self._check_common(dot_prod_attention)
+
+    def _test_sdpa_rewriter_12(self):
+        def dot_prod_attention(
+            query: torch.Tensor, key: torch.Tensor, value: torch.Tensor
+        ) -> torch.Tensor:
+            """Input tensors assumed to have shape (batch_size, seq_len, n_head, embed_dim)"""
+            q = query.transpose(1, 2)
+            k = key.transpose(1, 2)
+            v = value.transpose(1, 2)
+            return torch.nn.functional.dropout(
+                torch.matmul(q, k.transpose(-2, -1))
+                .div(math.sqrt(key.shape[-1]))
+                .softmax(dim=-1)
+                .matmul(v),
+                p=0.4,
+                training=True,
+                inplace=False,
+            )
+
+        self._check_common(dot_prod_attention, contains=False, has_dropout=True)
+
+
+if HAS_CUDA and PLATFORM_SUPPORTS_FUSED_SDPA:
+
+    class SDPAPatternRewriterCudaTests(TestSDPAPatternRewriterTemplate):
+        device = "cuda"
+        test_sdpa_rewriter_1_cuda = (
+            TestSDPAPatternRewriterTemplate._test_sdpa_rewriter_1
```

Comment:
This is a more concise, canonical way of achieving this

https://github.com/pytorch/pytorch/blob/main/test/inductor/test_inductor_freezing.py#L654-L670